tests/unit/ResolutionDates.spec.ts

```diff
@@ -0,0 +1,250 @@
+
+    // verify add button
+    const button = cols.at(2).find('.add-btn')
+    expect(button).toBeDefined()
+    expect(button.text()).toBe('Add')
+
+    wrapper.destroy()
+  })
+
+  it('displays previous dates', async () => {
+    const wrapper = wrapperFactory({ previousDates: arrayOfDates })
+    const vm = wrapper.vm
+
+    // verify there is a second row
+    const rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(2)
+
+    // verify second row has 2 columns
+    const cols = rows.at(1).findAll('.col')
+    expect(cols.length).toBe(2)
+
+    // verify Show button
+    const button = cols.at(1).find('.show-previous-dates-btn')
+    expect(button).toBeDefined()
+    expect(button.text()).toBe('Show previous resolutions or court order dates')
+
+    // click the button
+    await vm.$el.querySelector('.show-previous-dates-btn').click()
+
+    // verify Hide button
+    expect(button.text()).toBe('Hide previous resolutions or court order dates')
+
+    // verify previous dates
+    const items = cols.at(1).findAll('li')
+    expect(items.length).toBe(2)
+    expect(items.at(0).text()).toBe('2020-05-23')
+    expect(items.at(1).text()).toBe('2020-06-01')
+
+    // click the button again
+    await vm.$el.querySelector('.show-previous-dates-btn').click()
+
+    // verify that it went back to the original text
+    expect(button.text()).toBe('Show previous resolutions or court order dates')
+
+    wrapper.destroy()
+  })
+
+  it('displays added dates', () => {
+    const wrapper = wrapperFactory({ addedDates: arrayOfDates })
+    const vm = wrapper.vm
+
+    // verify there is a second row
+    const rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(2)
+
+    // verify second row has 2 columns
+    const cols = rows.at(1).findAll('.col')
+    expect(cols.length).toBe(2)
+
+    // verify added dates and buttons
+    const items = cols.at(1).findAll('li')
+    expect(items.length).toBe(2)
+    expect(items.at(0).text()).toContain('2020-05-23')
+    expect(items.at(1).text()).toContain('2020-06-01')
+
+    // verify Remove buttons
+    let button = items.at(0).find('.remove-btn')
+    expect(button).toBeDefined()
+    expect(button.text()).toBe('Remove')
+    button = items.at(1).find('.remove-btn')
+    expect(button).toBeDefined()
+    expect(button.text()).toBe('Remove')
+
+    wrapper.destroy()
+  })
+
+  it('can add a date', async () => {
+    const wrapper = wrapperFactory()
+    const vm = wrapper.vm
+
+    // verify there is just 1 row
+    let rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(1)
+
+    // click the Add button
+    await vm.$el.querySelector('.add-btn').click()
+
+    // verify there are now 2 rows
+    rows = wrapper.findAll('.row')
+    expect(rows.length).toBe(2)
+
+    // verify date picker form
+    expect(wrapper.find('.date-picker-form')).toBeDefined()
+
+    // inject a date
+    vm.onDate('2021-03-17')
```

Comment:
It's not this component's job to test the date picker, so I simply call the event handler here.